### PR TITLE
Verifies if the ConfigMap object and the expected fields exists

### PR DESCRIPTION
**Pull request by fao89**

### CODE REVIEW

**Review by SeanMooney — COMMENTED** [review]
cool this mirrors the secret one so once we are happy with that we can merge this as well.

hopefully we will evenrutlay have envtest coverage for this in the future.

File: modules/common/configmap/configmap.go [hunk 1]
```diff
@@ -263,3 +263,42 @@ func GetConfigMap(
...
+	reader client.Reader,
+	requeueTimeout time.Duration,
+) (string, ctrl.Result, error) {
+	configMap := &corev1.ConfigMap{}
+	err := reader.Get(ctx, configMapName, configMap)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			return "",
+				ctrl.Result{RequeueAfter: requeueTimeout},
+				fmt.Errorf("OpenStack ConfigMap %s not found", configMapName)
```
Comment (gibizer) [inline modules/common/configmap/configmap.go:282]: please drop OpenStack here

**Review by gibizer — CHANGES_REQUESTED** [review]

**Review by gibizer — APPROVED** [review]
looks good to me

**Review by SeanMooney — APPROVED** [review]
this also looks good to me

**Review by stuggi — APPROVED** [review]
/lgtm Thanks!

**Pull request merged**